### ISSUE TITLE
Receiving "error: video_input_init: Bad scale conversion type" and black screen

### ISSUE THREAD

**Issue by voxxit** [johnboiles/obs-mac-virtualcam#45]
Not sure what is up, though. Here is the output of my log:

```build ❯ cd $OBS_DIR/build/rundir/RelWithDebInfo/bin                                                              master
./obs
info: CPU Name: Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz
info: CPU Speed: 3200MHz
info: Physical Cores: 6, Logical Cores: 12
info: Physical Memory: 65536MB Total
info: OS Name: Mac OS X (NSMACHOperatingSystem)
info: OS Version: Version 10.15.4 (Build 19E287)
info: Kernel Version: 19.4.0
info: hotkeys-cocoa: Using layout 'com.apple.keylayout.US'
info: Portable mode: false
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_clicked()
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_destroyed()
QMetaObject::connectSlotsByName: No matching signal for on_actionGridMode_triggered()
QMetaObject::connectSlotsByName: No matching signal for on_program_customContextMenuRequested(QPoint)
info: OBS 25.0.7-48-g824a0ec4 (mac)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
	samples per sec: 44100
	speakers:        2
info: ---------------------------------
info: Initializing OpenGL...
info: Loading up OpenGL on adapter ATI Technologies Inc. AMD Radeon Pro 580 OpenGL Engine
info: OpenGL loaded successfully, version 4.1 ATI-3.8.24, shading language 4.10
info: ---------------------------------
info: video settings reset:
	base resolution:   1080x1920
	output resolution: 1080x1920
	downscale filter:  Bilinear
	fps:               30/1
	format:            NV12
	YUV mode:          709/Full
info: NV12 texture support not available
info: Audio monitoring device:
	name: Default
	id: default
info: ---------------------------------
warning: Failed to load 'en-US' text for module: 'decklink-ouput-ui.so'
error: os_dlopen(libpython3.7m.dylib->libpython3.7m.dylib): dlopen(libpython3.7m.dylib, 257): image not found

warning: [Python] Could not load library: libpython3.7m.dylib
CMIOObjectID:	0x23
	Class:		CMIO Stream
	Name:		Sample Stream
	Direction:	Input
	Channels:	1
CMIOObjectID:			0x22
	Class:				CMIO Device
	Name:				ManyCam Virtual Webcam
	Input Channels:		1
	Output Channels:	0
CMIOObjectID:			0x21
	Class:				CMIO PlugIn
	Name:				com.visicom.ManyCam.VideoDevice.plugin
info: Decklink API Compiled version 10.11.4
info: Decklink API Installed version 11.5
2020-04-23 18:56:12.385 obs[37500:3256912] apply_selection_policy_once: prefer use of removable GPUs (via (null):GPUSelectionPolicy->preferRemovable)
info: [VideoToolbox encoder]: Adding VideoToolbox H264 encoders
warning: Failed to load 'en-US' text for module: 'obs-mac-virtualcam.so'
error: os_dlopen(/Library/Application Support/obs-studio/plugins/StreamDeckPlugin/bin/StreamDeckPlugin.so->/Library/Application Support/obs-studio/plugins/StreamDeckPlugin/bin/StreamDeckPlugin.so): dlopen(/Library/Application Support/obs-studio/plugins/StreamDeckPlugin/bin/StreamDeckPlugin.so, 257): Library not loaded: @executable_path/../Frameworks/libobs.0.dylib
  Referenced from: /Library/Application Support/obs-studio/plugins/StreamDeckPlugin/bin/StreamDeckPlugin.so
  Reason: image not found

warning: Module '/Library/Application Support/obs-studio/plugins/StreamDeckPlugin/bin/StreamDeckPlugin.so' not loaded
info: Loading iOS Camera Plugin (version 2.5.1)
info: [obs-browser]: Version 2.7.1
info: mkdir locale dist path is  /Users/joshdelsman/Library/Application Support/obs-studio/dist
info: [Lovese] websocket_server_start !
error: os_dlopen(/Library/Application Support/obs-studio/plugins/obs-ndi/bin/obs-ndi.so->/Library/Application Support/obs-studio/plugins/obs-ndi/bin/obs-ndi.so): dlopen(/Library/Application Support/obs-studio/plugins/obs-ndi/bin/obs-ndi.so, 257): Library not loaded: @rpath/QtWidgets
  Referenced from: /Library/Application Support/obs-studio/plugins/obs-ndi/bin/obs-ndi.so
  Reason: image not found

warning: Module '/Library/Application Support/obs-studio/plugins/obs-ndi/bin/obs-ndi.so' not loaded
error: os_dlopen(/Library/Application Support/obs-studio/plugins/obs-websocket/bin/obs-websocket.so->/Library/Application Support/obs-studio/plugins/obs-websocket/bin/obs-websocket.so): dlopen(/Library/Application Support/obs-studio/plugins/obs-websocket/bin/obs-websocket.so, 257): Library not loaded: @rpath/QtWidgets
  Referenced from: /Library/Application Support/obs-studio/plugins/obs-websocket/bin/obs-websocket.so
  Reason: image not found

warning: Module '/Library/Application Support/obs-studio/plugins/obs-websocket/bin/obs-websocket.so' not loaded
info: ---------------------------------
info:   Loaded Modules:
info:     obs-lovense-toolset.so
info:     obs-ios-camera-source.so
info:     lovense-tools-update.so
info:     text-freetype2.so
info:     rtmp-services.so
info:     obs-x264.so
info:     obs-vst.so
info:     obs-transitions.so
info:     obs-outputs.so
info:     obs-mac-virtualcam.so
info:     obs-filters.so
info:     obs-ffmpeg.so
info:     mac-vth264.so
info:     mac-syphon.so
info:     mac-decklink.so
info:     mac-capture.so
info:     mac-avcapture.so
info:     image-source.so
info:     frontend-tools.so
info:     decklink-ouput-ui.so
info:     coreaudio-encoder.so
info: ---------------------------------
info: ==== Startup complete ===============================================
info: [Lovense-Update]{"hasUpdate":false,"force":false,"version":"","url":""}
info: All scene data cleared
info: ------------------------------------------------
error: Source ID 'browser_source' not found
error: Failed to create source 'Browser'!
info: Switched to scene 'Scene'
info: ------------------------------------------------
info: Loaded scenes:
info: - scene 'Scene':
info:     - source: 'Browser' (browser_source)
info: ------------------------------------------------
info: Switched to Preview/Program mode
info: ------------------------------------------------
info: Settings changed (stream 1)
info: ------------------------------------------------
info: ---------------------------------
info: [x264 encoder: 'streaming_h264'] preset: ultrafast
info: [x264 encoder: 'streaming_h264'] profile: high
info: [x264 encoder: 'streaming_h264'] settings:
	rate_control: CBR
	bitrate:      6000
	buffer size:  6000
	crf:          0
	fps_num:      30
	fps_den:      1
	width:        1080
	height:       1920
	keyint:       60

info: [CoreAudio AAC: 'avc_aac_stream']: settings:
	mode:          AAC
	bitrate:       160
	sample rate:   44100
	cbr:           on
	output buffer: 1536
info: [rtmp stream: 'adv_stream'] Connecting to RTMP URL rtmp://127.0.0.1/live...
info: [rtmp stream: 'adv_stream'] Connection to rtmp://127.0.0.1/live successful
info: ==== Streaming Start ===============================================
info: ==== Recording Start ===============================================
info: Output 'adv_ffmpeg_output': stopping
info: Output 'adv_ffmpeg_output': Total frames output: 1143
info: Output 'adv_ffmpeg_output': Total drawn frames: 1150
info: ==== Recording Stop ================================================
info: Switched to scene 'Scene'
info: Switched to regular Preview mode
info: ------------------------------------------------
info: [rtmp stream: 'adv_stream'] User stopped the stream
info: Output 'adv_stream': stopping
info: Output 'adv_stream': Total frames output: 1274
info: Output 'adv_stream': Total drawn frames: 1294
info: ==== Streaming Stop ================================================
info: ---------------------------------
info: video settings reset:
	base resolution:   1920x1080
	output resolution: 1920x1080
	downscale filter:  Bilinear
	fps:               60/1
	format:            NV12
	YUV mode:          709/Full
info: NV12 texture support not available
info: Settings changed (outputs, video)
info: ------------------------------------------------
info: Video Capture Device: No device selected
info: User added source 'Video Capture Device' (av_capture_input) to scene 'Scene'
info: Video Capture Device: Selected device 'Blackmagic UltraStudio Mini Recorder'
info: Video Capture Device: Using preset 1280x720
warning: Video Capture Device: Could not load resolution
warning: Video Capture Device: Could not load frame rate
info: Video Capture Device: No device selected, stopping capture
info: User Removed source 'Video Capture Device' (av_capture_input) from scene 'Scene'
info: User Removed source 'Browser' (browser_source) from scene 'Scene'
info: User added source 'Blackmagic Device' (decklink-input) to scene 'Scene'
info: decklink: Starting capture...
error: decklink: Failed to enable video input
info: decklink: Starting capture...
error: decklink: Failed to enable video input
info: decklink: Starting capture...
error: decklink: Failed to enable video input
info: decklink: Starting capture...
info: adding 23 milliseconds of audio buffering, total audio buffering is now 23 milliseconds (source: Blackmagic Device)

info: decklink: Stopping capture of 'UltraStudio Mini Recorder'...
info: decklink: Starting capture...
info: adding 23 milliseconds of audio buffering, total audio buffering is now 46 milliseconds (source: Blackmagic Device)

error: video_input_init: Bad scale conversion type
info: ---------------------------------
info: [x264 encoder: 'streaming_h264'] preset: ultrafast
info: [x264 encoder: 'streaming_h264'] profile: high
info: [x264 encoder: 'streaming_h264'] settings:
	rate_control: CBR
	bitrate:      6000
	buffer size:  6000
	crf:          0
	fps_num:      60
	fps_den:      1
	width:        1920
	height:       1080
	keyint:       120

info: [CoreAudio AAC: 'avc_aac_stream']: settings:
	mode:          AAC
	bitrate:       160
	sample rate:   44100
	cbr:           on
	output buffer: 1536
info: [rtmp stream: 'adv_stream'] Connecting to RTMP URL rtmp://127.0.0.1/live...
info: [rtmp stream: 'adv_stream'] Connection to rtmp://127.0.0.1/live successful
info: ==== Streaming Start ===============================================
info: ==== Recording Start ===============================================
info: Output 'virtualcam_output': stopping
info: Output 'virtualcam_output': Total frames output: 0
info: Output 'virtualcam_output': Total drawn frames: 7588 (7605 attempted)
info: Output 'virtualcam_output': Number of lagged frames due to rendering lag/stalls: 17 (0.2%)
error: video_input_init: Bad scale conversion type
info: Video Capture Device: No device selected
info: User added source 'Video Capture Device' (av_capture_input) to scene 'Scene'
info: Video Capture Device: Selected device 'C922 Pro Stream Webcam'
info: Video Capture Device: Using preset 1280x720
warning: Video Capture Device: Could not load resolution
warning: Video Capture Device: Could not load frame rate
info: Video Capture Device: Capturing 'C922 Pro Stream Webcam' (0x2230000046d085c):
	Resolution: 1920x1080
	FPS: 30 (30000030/1000000)
	Frame interval: 0.0333333 s
	Input format: Auto (actual: MJPEG - JPEG OpenDML (dmb1))
	Requested color space: Auto (-1)
	Requested video range: Auto (-1)
	Using format: <AVCaptureDeviceFormat: 0x7fdc82bbeda0> 'vide'/'dmb1' enc dims = 1920x1080, pres dims = 1920x1080 { 30.00 24.00 20.00 15.00 10.00 7.50 5.00 fps }
info: [rtmp stream: 'adv_stream'] User stopped the stream
info: Output 'adv_stream': stopping
info: Output 'adv_stream': Total frames output: 11677
info: Output 'adv_stream': Total drawn frames: 11669 (11706 attempted)
info: Output 'adv_stream': Number of lagged frames due to rendering lag/stalls: 37 (0.3%)
info: ==== Streaming Stop ================================================
info: Output 'adv_ffmpeg_output': stopping
info: Output 'adv_ffmpeg_output': Total frames output: 11732
info: Output 'adv_ffmpeg_output': Total drawn frames: 11704 (11741 attempted)
info: Output 'adv_ffmpeg_output': Number of lagged frames due to rendering lag/stalls: 37 (0.3%)
info: ==== Recording Stop ================================================
info: Output 'virtualcam_output': stopping
info: Output 'virtualcam_output': Total frames output: 0
info: Output 'virtualcam_output': Total drawn frames: 10096 (10133 attempted)
info: Output 'virtualcam_output': Number of lagged frames due to rendering lag/stalls: 37 (0.4%)
info: ---------------------------------
info: video settings reset:
	base resolution:   1920x1080
	output resolution: 1920x1080
	downscale filter:  Bilinear
	fps:               30/1
	format:            NV12
	YUV mode:          709/Full
info: NV12 texture support not available
info: Settings changed (video)
info: ------------------------------------------------
error: video_input_init: Bad scale conversion type
info: ---------------------------------
info: [x264 encoder: 'streaming_h264'] preset: ultrafast
info: [x264 encoder: 'streaming_h264'] profile: high
info: [x264 encoder: 'streaming_h264'] settings:
	rate_control: CBR
	bitrate:      6000
	buffer size:  6000
	crf:          0
	fps_num:      30
	fps_den:      1
	width:        1920
	height:       1080
	keyint:       60

info: [CoreAudio AAC: 'avc_aac_stream']: settings:
	mode:          AAC
	bitrate:       160
	sample rate:   44100
	cbr:           on
	output buffer: 1536
info: [rtmp stream: 'adv_stream'] Connecting to RTMP URL rtmp://127.0.0.1/live...
info: [rtmp stream: 'adv_stream'] Connection to rtmp://127.0.0.1/live successful
info: ==== Streaming Start ===============================================
info: ==== Recording Start ===============================================```

**Comment by voxxit** [comment]
So, just an update: I was able to get it to work eventually. It seems like the culprit is this bit of code:

https://github.com/obsproject/obs-studio/blob/c938ea712bce0e9d8e0cf348fd8f77725122b9a5/libobs/media-io/video-io.c#L315-L324

This led me to look at my Video options. I adjusted the Downscale Filter from Bicubic to Bilinear and restarted the virtual camera, and still nothing. So, then I just restarted OBS (leaving it at Bilinear) and it began to work 🤷‍♂️ 

Another piece of information, I did originally have it set to 1080p60... don't know if that contributed to the bad state.

**Comment by gxalpha** [comment]
Hi, that happened to me sometimes as well, usually after changing some video settings. In my experience, simply restarting obs fixed the issue. Did you try that before changing the downscale filters?

**Comment by johnboiles** [comment]
@voxxit thanks for the report. I _think_ I've run into that once or twice as well but I didn't investigate too deeply. I wonder if there's anything to do in the plugin to fix this

**Comment by johnboiles** [comment]
I think #83 is probably the same as this.

**Comment by johnboiles** [comment]
I think I fixed this in #147. Could y'all test it out by downloading the latest `pkg` build [here](https://github.com/johnboiles/obs-mac-virtualcam/actions/runs/116484810). Start a new movie recording in quicktime, start/stop the virtualcam a few times, maybe change the resolution in between, and let me know if you get this error.

**Comment by johnboiles** [comment]
I bet this is fixed. Going to close but please reopen if you encounter this when running 1.2.0
